src/main/java/org/zalando/nakadi/controller/EventStreamController.java

```diff
@@ -237,7 +243,17 @@ public StreamingResponseBody streamEvents(
 
                 outputStream.flush(); // Flush status code to client
 
-                eventStream.streamEvents(connectionReady);
+                eventStream.streamEvents(connectionReady, () -> {
+                    if (needCheckAuthorization.getAndSet(false)) {
+                        // in a case of exception all exceptions are treated in the same manner - all are internal
+                        // errors
+                        try {
+                            authorizationValidator.authorizeStreamRead(eventTypeRepository.findByName(eventTypeName));
+                        } catch (final InternalNakadiException | NoSuchEventTypeException ex) {
+                            throw new ServiceTemporarilyUnavailableException(ex);
+                        }
+                    }
+                });
```

Comment:
I don't really like that authorization now is spread through the controller. Is it somehow possible to compact it and keep in one place?